x-pack/plugins/enterprise_search/server/routes/app_search/credentials.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { IRouteDependencies } from '../../plugin';
+import { createEnterpriseSearchRequestHandler } from '../../lib/enterprise_search_request_handler';
+
+interface ICredentialsResponse {
+  results: object[];
+  meta?: {
+    page?: {
+      total_results: number;
```

Comment:
It might also be nice to add `current`, `total_pages`, and `size` here as well (which we might DRY out to a shared type, if all our other API endpoints have a similar meta obj), unless we really just want this definition to be for the sake of the hasValidData func

Reply:
https://github.com/elastic/kibana/pull/75487/commits/9f742e1955d3720f9c014f1d87aba279753963f8